### PR TITLE
Updating to 2021.1.14f1

### PR DESCRIPTION
**Pull request by Bod9001**
Updated mirror in the process,
Updated ignorance 

Note, I'll need some Help finding where the version numbers are in the Documents

### CODE REVIEW

**Comment by Bod9001** [comment]
humm, going to need some help to get the tests working?

**Comment by corp-0** [comment]
how critical is this update? Do we get any remarkable benefit from  updating?

**Comment by jamessimo** [comment]
> how critical is this update? Do we get any remarkable benefit from updating?

It may allow us to build to OSX BigSur since Unity stated they have made some fixes to it

**Comment by MaxIsJoe** [comment]
> how critical is this update? Do we get any remarkable benefit from updating?

I wouldn't call this update critical to our project but it is very welcome as it gives us : 

- If we ever want to utilize any addons that use URP we'd be much closer to all the changes Unity has made so it would save a lot of hassle of upgrading 2020 URP to 2021 URP and vice versa

- Performance Improvements

- Lots of crash fixes for the Unity Editor

- Minor Linux and MacOS fixes

**Comment by Bod9001** [comment]
yeah, though I need help with getting PR tests working, Also the Enter play mode time has been reduced slightly that's good

**Comment by NoooneyDude** [comment]
Why is the test results zip empty? Same for anyone else?

**Comment by Bod9001** [comment]
it's because it's complaining about not finding the right Unity version

**Comment by Bod9001** [comment]
player sprites need fixing

File: UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPartSprites.cs [hunk 1]
```diff
@@ -39,13 +39,18 @@ public void UpdateHideDlags( ClothingHideFlags newOne)
...
 		Data = InNew;
 		if (CustomNetworkManager.Instance._isServer)
 		{
 			ParentContainer.UpdateThis();
 			return;
 		}
 		SpriteOrder = JsonConvert.DeserializeObject<SpriteOrder>(Data);
+		if (SpriteOrder == null)
+		{
+			Logger.Log("o3o");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPartSprites.cs]: bod, that is a terrible log

Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPartSprites.cs]: haha was using for testing

**Comment by mmails** [comment]
ok but noone commented on the 6999

**Comment by Bod9001** [comment]
#6969 is more nice

**Comment by mmails** [comment]
> #6969 is more nice

Nice.

**Comment by Aranclanos** [comment]
deleted library and pulled from this branch on latest develop and there were a lot of errors when starting a server, seemed unplayable.
I'll test again from a new git clone

**Comment by Bod9001** [comment]
the only errors I get are ore generation being missing and Fox tails  having double sprites

**Comment by Aranclanos** [comment]
image
this time I cloned from your branch, didn't update to develop, same thing happened

**Comment by corp-0** [comment]
did you regenerate library folder?

**Comment by Bod9001** [comment]
I think thatdan had similar issue, they fixed it by restarting the editor I think?

**Comment by Aranclanos** [comment]
I updated to the last commit from 30mins ago and it seemed to work afterwards, trying out builds now

image
downed humans have their underclothes still vertical

image
ghosting and returning to your body seems to break your entire UI, you're unable to move. On clients it turned your screen dark.

restarting the round seemed to disconnect my client too

**Comment by MaxIsJoe** [comment]
> downed humans have their underclothes still vertical

You know, this wouldn't be an issue if we just rotated the player's root object instead of individually going through each and every sprite that's on the player to rotate it.

**Comment by corp-0** [comment]
> > downed humans have their underclothes still vertical
> 
> You know, this wouldn't be an issue if we just rotated the player's root object instead of individually going through each and every sprite that's on the player to rotate it.

No, rotating the root is stoopid. There is a game object that has all sprites as children.

**Comment by MaxIsJoe** [comment]
> No, rotating the root is stoopid. There is a game object that has all sprites as children.

Now correct me If I'm wrong but :
1. Last time I checked, body parts and customizations are completely separate from another and are not under any root "sprite manager" object.
2. Do you really, reeeaaallly think that having to manually rotate every new object's sprites we add on the player via code is not stoopid? Especially when they don't fall into any gameObject that does this automatically (like the floating action text)? It just seems like more work that's more prone to bugs in the future against a simpler solution that is 100% effective and does not cause major issues.

**Comment by PetMudstone** [comment]
>    Last time I checked, body parts and customizations are completely separate from another and are not under any root "sprite manager" object.

The clothing overlays and body part parents do already have objects they're under in the hierarchy on the V4 player prefab, notably.

image

**Comment by MaxIsJoe** [comment]
> > Last time I checked, body parts and customizations are completely separate from another and are not under any root "sprite manager" object.
> 
> The clothing overlays and body part parents do already have objects they're under in the hierarchy on the V4 player prefab, notably.
> 
> image

the bodypart component spawns under sprites?

**Comment by Bod9001** [comment]
All that needs to be done fix round restart kick and fix a ghost

**Comment by NoooneyDude** [comment]
> > downed humans have their underclothes still vertical
> 
> You know, this wouldn't be an issue if we just rotated the player's root object instead of individually going through each and every sprite that's on the player to rotate it.

Can't manipulate a gameobject that has CNT / PlayerSync.

**Pull request merged**